src/Estimators/WalkerTraceManager.h

```diff
@@ -0,0 +1,465 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2024 QMCPACK developers.
+//
+// File developed by: Jaron T. Krogel, krogeljt@ornl.gov, Oak Ridge National Laboratory
+//
+// File created by: Jaron T. Krogel, krogeljt@ornl.gov, Oak Ridge National Laboratory
+//////////////////////////////////////////////////////////////////////////////////////
+
+
+#ifndef QMCPLUSPLUS_WALKERTRACEMANAGER_H
+#define QMCPLUSPLUS_WALKERTRACEMANAGER_H
+
+
+#include <Configuration.h>
+#include "OhmmsPETE/OhmmsArray.h"
+#include "Message/Communicate.h"
+#include "hdf/hdf_archive.h"
+
+#include <unordered_set>
+
+
+
+namespace qmcplusplus
+{
+
+const unsigned int WDMAX = 4;
+using WTraceInt          = long;
+using WTraceReal         = OHMMS_PRECISION_FULL;
+using WTraceComp         = std::complex<WTraceReal>;
+#ifndef QMC_COMPLEX
+using WTracePsiVal = WTraceReal;
+#else
+using WTracePsiVal = WTraceComp;
+#endif
+
+
+
+struct WalkerQuantityInfo
+{
+  std::string name;
+  size_t dimension;
+  size_t size;
+  size_t unit_size;
+  TinyVector<size_t, WDMAX> shape;
+  size_t buffer_start;
+  size_t buffer_end;
+
+  WalkerQuantityInfo(const std::string& name_,size_t unit_size_,size_t buffer_start_,size_t n1=1,size_t n2=0,size_t n3=0,size_t n4=0)
+  {
+    name         = name_;
+    unit_size    = unit_size_;
+    buffer_start = buffer_start_;
+    shape[0]     = n1;
+    shape[1]     = n2;
+    shape[2]     = n3;
+    shape[3]     = n4;
```

Comment:
I don't get what are 0, 1, 2, 3. You can probably give a name to each index.
```
enum
{ DIM_0_NAME = 0,
  DIM_1_NAME,
  DIM_2_NAME,
  DIM_3_NAME,
  WDMAX }
```
WDMAX will be set 4 by the compiler.

Reply:
They don't have names, but I can give them names if you prefer.  They literally are dimension 0, 1, 2, 3 of an array.

Reply:
Done